### PR TITLE
Enable scala 3 pipeline

### PR DESCRIPTION
**Pull request by regiskuckaertz**

### CODE REVIEW

**Comment by regiskuckaertz** [comment]
There's a compilation error due to this compiler bug: https://github.com/lampepfl/dotty/issues/15855

**Comment by thesamet** [comment]
Is it only impact Scala.js on Scala 3? If so, we can exclude for the time being.

**Comment by regiskuckaertz** [comment]
Done (I found the flaky test, will work on this separately). I split the JVM and JS tests into two steps, so that I can exclude Scala3+JS but let me know if there is a better way of expressing the same thing—I've not written much GH actions.

File: e2e/src/test/scala/scalapb/zio_grpc/ClientCallsSpec.scala [hunk 1]
```diff
@@ -4,34 +4,40 @@ import io.grpc.{CallOptions, ManagedChannelBuilder}
...
-                      ),
-                      scalapb.zio_grpc.testservice.TestServiceGrpc.METHOD_UNARY,
-                      CallOptions.DEFAULT,
-                      meta,
-                      Request(Request.Scenario.DELAY, in = 12)
-                    )
-                    .retry(Schedule.recurs(2))
-                    .exit
+          channel <- ZIO
+                       .acquireRelease(
```
Comment (thesamet) [inline e2e/src/test/scala/scalapb/zio_grpc/ClientCallsSpec.scala:18]: Is there something we can factor out of this into the companion object of ZChannel to make it easier for the users to create a scoped channel?

Comment (regiskuckaertz) [reply e2e/src/test/scala/scalapb/zio_grpc/ClientCallsSpec.scala:18]: Sure, let me work this out.

Comment (regiskuckaertz) [reply e2e/src/test/scala/scalapb/zio_grpc/ClientCallsSpec.scala]: @thesamet done

**Comment by regiskuckaertz** [comment]
@thesamet gentle ping.

**Comment by thesamet** [comment]
Thank you!

**Pull request merged**